Module4_HM4_EF2/Module4_HM4_EF2/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Module4_HM4_EF2.DTOs;
+using Module4_HM4_EF2.Services.Abstractions;
+
+namespace Module4_HM4_EF2.Controllers
+{
+    [ApiController]
+    [Route("api/categories")]
+    public class CategoryController : ControllerBase
+    {
+        private readonly ICategoryService _categoryService;
+
+        public CategoryController(ICategoryService categoryService)
+        {
+            _categoryService = categoryService;
+        }
+
+        [HttpGet("name")]
```

Comment:
Usually, it is one get endpoint, with different queries, and then it delegates the search to the correct service/query. Because categories/name?name='hi' is not in a rest style.